includes/Modules/Analytics_4/Conversion_Reporting/Conversion_Reporting_Events_Sync.php

```diff
@@ -73,9 +87,18 @@ public function sync_detected_events() {
 			return;
 		}
 
+		// Get current stored detected events.
+		$settings              = $this->settings->get();
+		$saved_detected_events = isset( $settings['detectedEvents'] ) ? $settings['detectedEvents'] : array();
+
 		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
 		if ( empty( $report->rowCount ) ) {
 			$this->settings->merge( array( 'detectedEvents' => array() ) );
+			$this->transients->set( 'googlesitekit_conversion_reporting_detected_events', array() );
```

Comment:
Also it will be easier to convert these string slugs to a class constant, so it easier accessible from one place (and also from outside of class if needed sometimes in the future, or for tests) instead of using the same string in multiple places, for example `const DETECTED_EVENTS_TRANSIENT = 'googlesitekit_conversion_reporting_detected_events';` and same for the lost events.

So we can easily access them, eq `$this->transients->get( self::DETECTED_EVENTS_TRANSIENT ) `